FileNotFoundException when reinitialize table inside conditional steps/composite steps

**Scenario:**
```
When I initialize SCENARIO variable `table` with values:
|column |
|value  |
When the condition `#{eval( "true" )}` is true I do
|step                                          |
|When I run composite step with table:${table} |
```
**Composite:**
```
Composite: When I run composite step with table:$table
When I initialize SCENARIO variable `table` with values:<table>
```
**Actual:**
Failiing on ```When I initialize SCENARIO variable `table` with values:<table>```
```
java.io.FileNotFoundException: class path resource [[{column=value}]] cannot be opened because it does not exist

java.lang.IllegalArgumentException: java.io.FileNotFoundException: class path resource [[{column=value}]] cannot be opened because it does not exist
	at org.vividus.bdd.StoryLoader.resourceAsStream(StoryLoader.java:57)
	at org.jbehave.core.io.LoadFromClasspath.loadResourceAsText(LoadFromClasspath.java:66)
	at org.vividus.bdd.StoryLoader.loadResourceAsText(StoryLoader.java:45)
	at org.jbehave.core.model.ExamplesTableFactory.createExamplesTable(ExamplesTableFactory.java:81)
	at org.jbehave.core.steps.ParameterConverters$FunctionalParameterConverter.convertValue(ParameterConverters.java:599)
	at org.jbehave.core.steps.ParameterConverters$FunctionalParameterConverter.convertValue(ParameterConverters.java:584)
	at org.jbehave.core.steps.ParameterConverters.applyConverters(ParameterConverters.java:340)
	at org.jbehave.core.steps.ParameterConverters.convert(ParameterConverters.java:349)
	at org.vividus.bdd.steps.ParameterConvertersDecorator.convert(ParameterConvertersDecorator.java:77)
	at org.jbehave.core.steps.StepCreator.convertParameterValues(StepCreator.java:439)
	at org.jbehave.core.steps.StepCreator.access$1400(StepCreator.java:49)
	at org.jbehave.core.steps.StepCreator$ParametrisedStep.parametriseStep(StepCreator.java:889)
	at org.jbehave.core.steps.StepCreator$ParametrisedStep.perform(StepCreator.java:830)
	at org.jbehave.core.embedder.PerformableTree$FineSoFar.run(PerformableTree.java:390)
	at org.jbehave.core.embedder.PerformableTree$FineSoFar.run(PerformableTree.java:397)
	at org.vividus.bdd.steps.SubSteps.executeStep(SubSteps.java:76)
	at org.vividus.bdd.steps.SubSteps.execute(SubSteps.java:61)
	at org.vividus.bdd.steps.ExecutableSteps.performAllStepsIfConditionIsTrue(ExecutableSteps.java:75)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.jbehave.core.steps.StepCreator$ParametrisedStep.perform(StepCreator.java:834)
	at org.jbehave.core.embedder.PerformableTree$FineSoFar.run(PerformableTree.java:390)
	at org.jbehave.core.embedder.PerformableTree$PerformableSteps.perform(PerformableTree.java:1334)
	at org.jbehave.core.embedder.PerformableTree$AbstractPerformableScenario.perform(PerformableTree.java:1211)
	at org.jbehave.core.embedder.PerformableTree$AbstractPerformableScenario.performRestartableSteps(PerformableTree.java:1190)
	at org.jbehave.core.embedder.PerformableTree$AbstractPerformableScenario.performScenario(PerformableTree.java:1202)
	at org.jbehave.core.embedder.PerformableTree$NormalPerformableScenario.perform(PerformableTree.java:1249)
	at org.jbehave.core.embedder.PerformableTree$PerformableScenario.perform(PerformableTree.java:1105)
	at org.jbehave.core.embedder.PerformableTree$PerformableStory.performScenarios(PerformableTree.java:992)
	at org.jbehave.core.embedder.PerformableTree$PerformableStory.perform(PerformableTree.java:960)
	at org.jbehave.core.embedder.PerformableTree.performCancellable(PerformableTree.java:469)
	at org.jbehave.core.embedder.PerformableTree.perform(PerformableTree.java:436)
	at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:297)
	at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:270)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:830)
Caused by: java.io.FileNotFoundException: class path resource [[{column=value}]] cannot be opened because it does not exist
	at org.springframework.core.io.ClassPathResource.getInputStream(ClassPathResource.java:180)
	at org.vividus.bdd.StoryLoader.resourceAsStream(StoryLoader.java:53)
	... 39 more
```

--- a/build.gradle
+++ b/build.gradle
@@ -78,7 +78,7 @@ ext {
         hamcrest:               '2.2',
         httpclient:             '4.5.13',
         javaxInject:            '1',
-        jbehave:                '4.8.0-alpha.6',
+        jbehave:                '4.8.0-alpha.7',
         junit4:                 '4.13',
         slf4j:                  '2.0.0-alpha1',
         spring:                 '5.3.1',

--- a/vividus-tests/src/main/resources/steps/defaults/steps-for-integration-tests.steps
+++ b/vividus-tests/src/main/resources/steps/defaults/steps-for-integration-tests.steps
@@ -9,3 +9,6 @@ When I initialize the scenario variable `after-comment` with value `after`
 
 Composite: When I use composite step in known-issues
 Then `0` is = `1`
+
+Composite: When I run composite step with table:$table
+When I initialize scenario variable `table-from-composite-step` with values:<table>

--- a/vividus-tests/src/main/resources/story/integration/VariableTests.story
+++ b/vividus-tests/src/main/resources/story/integration/VariableTests.story
@@ -110,3 +110,13 @@ When I execute steps while counter is <= `1` with increment `1` starting from `1
 |!When I initialize the scenario variable `test2` with value `${iterationVariable}`!|
 |Then `${test2}` is equal to `1`                                                    |
 Then `${iterationVariable}` is equal to `42`
+
+Scenario: Pass ExamplesTable as variable to composite step accepting ExamplesTable
+Meta:
+    @issueId 1136
+When I initialize scenario variable `table` with values:
+|column |
+|value  |
+When I run composite step with table:${table}
+Then `${table[0].column}` is equal to `value`
+Then `${table-from-composite-step[0].column}` is equal to `value`

--- a/vividus/src/main/java/org/vividus/bdd/converter/ExamplesTableToListOfMapsConverter.java
+++ b/vividus/src/main/java/org/vividus/bdd/converter/ExamplesTableToListOfMapsConverter.java
@@ -1,0 +1,48 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.bdd.converter;
+
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.inject.Named;
+
+import org.apache.commons.lang3.reflect.TypeLiteral;
+import org.apache.commons.lang3.reflect.TypeUtils;
+import org.jbehave.core.model.ExamplesTable;
+import org.jbehave.core.steps.ParameterConverters.AbstractChainableParameterConverter;
+import org.jbehave.core.steps.Parameters;
+
+@Named
+public class ExamplesTableToListOfMapsConverter
+        extends AbstractChainableParameterConverter<ExamplesTable, List<Map<String, String>>>
+{
+    @Override
+    public boolean accept(Type type)
+    {
+        TypeLiteral<List<Map<String, String>>> typeLiteral = new TypeLiteral<>() { };
+        return TypeUtils.equals(type, typeLiteral.getType());
+    }
+
+    @Override
+    public List<Map<String, String>> convertValue(ExamplesTable examplesTable, Type type)
+    {
+        return examplesTable.getRowsAsParameters().stream().map(Parameters::values).collect(Collectors.toList());
+    }
+}

--- a/vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java
+++ b/vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java
@@ -25,13 +25,11 @@ import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.regex.Pattern;
-import java.util.stream.Collectors;
 
 import org.jbehave.core.annotations.Given;
 import org.jbehave.core.annotations.Then;
 import org.jbehave.core.annotations.When;
 import org.jbehave.core.model.ExamplesTable;
-import org.jbehave.core.steps.Parameters;
 import org.vividus.bdd.context.IBddVariableContext;
 import org.vividus.bdd.util.MapUtils;
 import org.vividus.bdd.variable.VariableScope;
@@ -232,11 +230,9 @@ public class BddVariableSteps
      */
     @When("I initialize $scopes variable `$variableName` with values:$examplesTable")
     public void initVariableWithGivenValues(Set<VariableScope> scopes, String variableName,
-                ExamplesTable examplesTable)
+            List<Map<String, String>> examplesTable)
     {
-        List<Map<String, String>> listOfMaps = examplesTable.getRowsAsParameters()
-                .stream().map(Parameters::values).collect(Collectors.toList());
-        bddVariableContext.putVariable(scopes, variableName, listOfMaps);
+        bddVariableContext.putVariable(scopes, variableName, examplesTable);
     }
 
     private void publishMapComparisonResults(List<List<EntryComparisonResult>> results)

--- a/vividus/src/test/java/org/vividus/bdd/converter/ExamplesTableToListOfMapsConverterTests.java
+++ b/vividus/src/test/java/org/vividus/bdd/converter/ExamplesTableToListOfMapsConverterTests.java
@@ -1,0 +1,73 @@
+/*
+ * Copyright 2019-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.bdd.converter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.reflect.TypeLiteral;
+import org.jbehave.core.model.ExamplesTable;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class ExamplesTableToListOfMapsConverterTests
+{
+    private final ExamplesTableToListOfMapsConverter converter = new ExamplesTableToListOfMapsConverter();
+
+    @Test
+    void shouldAcceptType()
+    {
+        assertTrue(converter.accept(new TypeLiteral<List<Map<String, String>>>() { }.getType()));
+    }
+
+    static Stream<Type> notAcceptableTypes()
+    {
+        return Stream.of(
+                new TypeLiteral<List<Map<String, Object>>>() { }.getType(),
+                new TypeLiteral<List<Map<Object, String>>>() { }.getType(),
+                new TypeLiteral<List<Map<?, ?>>>() { }.getType(),
+                new TypeLiteral<List<Set<String>>>() { }.getType(),
+                new TypeLiteral<List<String>>() { }.getType(),
+                new TypeLiteral<List<?>>() { }.getType(),
+                String.class
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("notAcceptableTypes")
+    void shouldNotAcceptType(Type type)
+    {
+        assertFalse(converter.accept(type));
+    }
+
+    @Test
+    void shouldConvertExamplesTable()
+    {
+        ExamplesTable table = new ExamplesTable("|key0|key1|\n|value0|value1|");
+        List<Map<String, String>> actual = converter.convertValue(table, null);
+        List<Map<String, String>> expected = List.of(Map.of("key0", "value0", "key1", "value1"));
+        assertEquals(expected, actual);
+    }
+}

--- a/vividus/src/test/java/org/vividus/bdd/steps/BddVariableStepsTests.java
+++ b/vividus/src/test/java/org/vividus/bdd/steps/BddVariableStepsTests.java
@@ -342,9 +342,8 @@ class BddVariableStepsTests
     void testInitVariableWithGivenValues()
     {
         Set<VariableScope> scopes = Set.of(VariableScope.SCENARIO);
-        ExamplesTable table = new ExamplesTable("|key0|key1|\n|value0|value1|");
-        bddVariableSteps.initVariableWithGivenValues(scopes, VALUE, table);
         List<Map<String, String>> listOfMaps = List.of(Map.of("key0", "value0", "key1", "value1"));
+        bddVariableSteps.initVariableWithGivenValues(scopes, VALUE, listOfMaps);
         verify(bddVariableContext).putVariable(scopes, VALUE, listOfMaps);
     }
 }

